libcaf_test/caf/test/approx.hpp

```diff
@@ -0,0 +1,54 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include <limits>
+#include <type_traits>
+
+namespace caf::test {
+
+/// Allows comparing a value of type `T` with a configurable precision.
+template <class T>
+class approx {
+public:
+  explicit approx(T value) : value_(value) {
+  }
+
+  approx epsilon(T eps) const {
+    auto copy = *this;
+    copy.epsilon_ = eps;
+    return copy;
+  }
+
+  friend bool operator==(T lhs, approx<T> rhs) {
+    if constexpr (std::is_floating_point_v<T>) {
+      return lhs >= rhs.min_value() && lhs <= rhs.max_value();
+    } else {
+      return lhs == rhs.value_;
+    }
```

Comment:
> we will get compilation error for types where the used operators (>=, <=, + and - ) are not supported.

What type would we pass to `approx` where this is a problem? Using `==` for the `else` branch is incorrect: `check_eq(5, approx{4}.epsilon(1)` should pass.